### DISCUSSION TITLE
Map workflow spec to AWS Step Functions

### DISCUSSION THREAD

**Discussion by drewburlingame** [serverlessworkflow/specification#582] [General]
My company is looking to implement a workflow service using AWS Step Functions and we're interested to see if we can use the workflow spec to define the spec and map that spec to Step Functions.

Discussed with @tsurdilo in [#serverless-workflow](https://cloud-native.slack.com/archives/CMBAZQSEP/p1643150809080000)

The first step is likely to map features of serverless workflow to step functions and identify discrepancies. @yzhao244 may have something to share for this.

**Comment by yzhao244** [comment]
@drewburlingame Here is the DSL comparison for ASL and SW. I did as much as I could at that time to compare amazon state language features with serverless workflows feature. Most of the features I think are quite aligned and exchangeable. However, our DSL still does differently and has more elements than ASL.. To me, ASL is more like a subset of SW DSL.   

One more thing, since I did this research early last year I remember, my comparisons may not reflect the latest specification of SW lol. We add more features since 0.5 such as better error handling, timeout and so on which I believe @tsurdilo knows better than me :) ... 

Anyways,  I hope this comparison can help us as a starting point. lol
[SW-AWS-DSL-Comparsion.xlsx](https://github.com/serverlessworkflow/specification/files/7952522/SW-AWS-DSL-Comparsion.xlsx)

**Reply by tsurdilo** [reply]
@yzhao244 wow this is awesome! thank you. would it be ok if i move it to markdown format and do a pr to our comparisons dir? that way we could do PRs against it.

**Reply by tsurdilo** [reply]
i can also add some other features of our DSL, the ones you mentioned, but also things like cron definition, compensation secrets, constants, etc

**Reply by yzhao244** [reply]
@tsurdilo Sure, please go ahead. :)
